esrally/mechanic/launcher.py

```diff
@@ -122,83 +127,32 @@ def stop(self, c):
-            if "error" in line.lower():
-                lines_to_log = 10
-            # don't log each output line as it is contained in the node's log files anyway and we just risk spamming our own log.
-            if not self.startup_event.isSet() or lines_to_log > 0:
-                self.logger.info("%s (stdout): %s", self.node_name, line)
-                lines_to_log -= 1
-
-            # no need to check as soon as we have detected node startup
-            if not self.startup_event.isSet():
-                if line.find("Initialization Failed") != -1 or line.find("A fatal exception has occurred") != -1:
-                    self.logger.error("[%s] encountered initialization errors.", self.node_name)
-                    # wait a moment to ensure the process has terminated before we signal that we detected a (failed) startup.
-                    self.await_termination(self.server)
-                    self.startup_event.set()
-                if line.endswith("started") and not self.startup_event.isSet():
-                    self.startup_event.set()
-                    self.logger.info("[%s] has successfully started.", self.node_name)
-
-    def await_termination(self, server, timeout=5):
-        # wait a moment to ensure the process has terminated
-        wait = timeout
-        while not server.returncode or wait == 0:
-            time.sleep(0.1)
-            server.poll()
-            wait -= 1
-
-
-def _start(process, node_name):
-    log = logging.getLogger(__name__)
-    startup_event = threading.Event()
-    watcher = StartupWatcher(node_name, process, startup_event)
-    t = threading.Thread(target=watcher.watch)
-    t.setDaemon(True)
-    t.start()
-    if startup_event.wait(timeout=InProcessLauncher.PROCESS_WAIT_TIMEOUT_SECONDS):
-        process.poll()
-        # has the process terminated?
-        if process.returncode:
-            msg = "Node [%s] has terminated with exit code [%s]." % (node_name, str(process.returncode))
-            log.error(msg)
-            raise exceptions.LaunchError(msg)
-        else:
-            log.info("Started node [%s] with PID [%s].", node_name, process.pid)
-            return process
-    else:
-        msg = "Could not start node [%s] within timeout period of [%s] seconds." % (
-            node_name, InProcessLauncher.PROCESS_WAIT_TIMEOUT_SECONDS)
-        # check if the process has terminated already
-        process.poll()
-        if process.returncode:
-            msg += " The process has already terminated with exit code [%s]." % str(process.returncode)
-        else:
-            msg += " The process seems to be still running with PID [%s]." % process.pid
-        log.error(msg)
-        raise exceptions.LaunchError(msg)
+    output = subprocess.check_output(args=shlex.split(compose_ps_cmd))
+    return output.decode("utf-8").rstrip()
+
+
+def _wait_for_healthy_running_container(container_id, timeout=60):
+    import docker
```

Comment:
I didn't originally put it in setup.py because I didn't want to impose the docker SDK requirement on people who weren't using Docker functionality.
However, since I am removing usage of the Docker SDK entirely, I will mark this as "resolved"